### ISSUE TITLE
Housekeeping transaction broadcast fails

### ISSUE THREAD

**Issue by bonomat** [itchysats/itchysats#844]
After restart, the maker printed an error that a certain transaction could not be published: 

```
2021-12-09 09:03:05 ERROR daemon::housekeeping: Broadcasting transaction failed. Txid: 3e3cf3f7a042e0532278b5d8ec72f6cefdb63a98250d5dd3513bf8d35629fa5f. Raw transaction: 020000000001011a2b51735a5b48d77c0f05cd1a7586729719b1e496946b71e1dd25ccfa5b66cc0000000000ffffffff015074040000000000220020be7ecdd38a810153ff3a31fbc14ca96807b245d355659b27de282c327c39573703483045022100facf87897a77b9e8c4ca7f128a276f02fffaf749166eea3f05ad851943deed0e0220247bd93c8189ec81ba37ea041d716a640eb2b771fd10e75602addbd1bb18ed6f01483045022100a72e1d13fbda429a991626a8200afa7a4075a3ccb0761ff344ac4e9f272a92f50220270f8940ea92a0bd2dae0af7df61157935e6587ea90234b5fc843dfdc4f8105801462102d3e13e04b8ca6fdfa16db929efce8617aadaa1cc35ba6e81a70c34079ec4377ead210384175418173da39e0803570e287258d53131c3e7316645e1b9e5f98dc777de43ac00000000: Electrum(Protocol(String("sendrawtransaction RPC error: {\"code\":-25,\"message\":\"bad-txns-inputs-missingorspent\"}")))

```

In this case, this was the `tx_commit` and it was expected as the taker published it in the meantime. 

We should check if the transaction is already in the mempool before trying to publishing it. If it is there, then everything is as expected, otherwise we can fail.

**Comment by thomaseizinger** [comment]
> We should check if the transaction is already in the mempool before trying to publishing it. If it is there, then everything is as expected, otherwise we can fail.

We do check that. In your case, it seems to not have been the same transaction? Your error code is `bad-txns-inputs-missingorspent`. AFAIK, there is a dedicated error code for "transaction already in blockchain".

**Comment by bonomat** [comment]
Could be that electrum is not as smart as we think? 
The transaction is in the blockchain: https://mempool.space/tx/3e3cf3f7a042e0532278b5d8ec72f6cefdb63a98250d5dd3513bf8d35629fa5f

**Comment by bonomat** [comment]
Closed in favor of #873
